scripts/docs/build-docs.ts

```diff
@@ -0,0 +1,117 @@
+import { join } from 'path';
+import { copy, mkdirp, remove, outputFile, move } from 'fs-extra';
+import { Clone, Checkout, Repository } from 'nodegit';
+
+import { generateGithubSpaScript } from './ghspa-template';
+import { runCommand } from './run-command';
+
+import { repository as REPO_URL, outDir as OUT_DIR } from './config.json';
+const WORK_DIR = join(process.cwd(), '../_DOCS_BUILD_WORK_DIR_');
+const DIST_DIR = join(WORK_DIR, 'dist');
+const MASTER_BRANCH_DIR = join(WORK_DIR, 'MASTER');
+const DOCS_VERSIONS_PATH = join(MASTER_BRANCH_DIR, 'docs/versions.json');
+
+interface VersionsConfig {
+  currentVersionName: string;
+  versions: Version[];
+}
+
+interface Version {
+  checkoutTarget: string;
+  name: string;
+  path: string;
+}
+
+(async function () {
```

Comment:
so hard to read this with all the `console.log` mess... 
should we make a `log` function to at least make the code a bit more readable?

Reply:
Done in b6a381cb62549ce30b2b0c3217b1452c3285ccf7